### PR TITLE
Add realtime-react

### PR DESCRIPTION
**Pull request by ankit-adapt**
# Summary

This PR adds the `realtime-react` package, which provides `useRealtime` hook for managing realtime connections. The `useRealtime` hook uses the `realtimeConnectionMachine` under the hood. You can explore and interact with the machine here: https://stately.ai/registry/editor/a9bcc4d7-6c9b-4fd5-b514-caeddcd6bd6f?machineId=7f470597-e52d-4a2f-b6a7-93333df86bb9&mode=Simulate

# Changes

1. Adds `useRealtime` hook to manage realtime connection
2. Adds `useAvailableMediaDevices` to get the list of available audio and video devices.
3. Updates RtAudio, RtAudioVisualizer, RtChat, RtVideo so that they are compatible with the new package.
4. Updates the landing page.

# Screenshot

image

# Testing

https://github.com/user-attachments/assets/481e9c9e-c527-42c2-ace6-2e06571ae46c

# Note

Before reviewing this PR, please review PR #2  first. Once the review is completed, merge this PR into `/ankitkumar/get-19-implementation-update-sdk`, and then merge PR #2 into the main branch.

The following tasks are still pending:

1. Add a context/provider to make the real-time connection available to all child components.
2. Write and add tests.
3. Update the documentation.
4. Cleanup

### CODE REVIEW

File: src/App.tsx
Comment (janak2) [inline src/App.tsx:1]: Is it possible to use Tailwind instead of custom CSS

File: src/App.tsx [hunk 1]
```diff
@@ -1,132 +1,19 @@
...
-  const { audioInput, videoInput, offerUrl, functionUrl } = values;
+import "./App.css";
 
-  const dumpConfigAndRun = () => {
-    const configDump = dump();
-    setConfig(configDump);
-    setIsRealtimeDisabled(false);
-  };
+export default function App() {
+  const [config, setConfig] = React.useState<TRealtimeConfig | null>(null);
```
Comment (janak2) [inline src/App.tsx]: I thought our plan was to abstract away the config?

File: src/App.tsx
Comment (ankit-adapt) [reply src/App.tsx:1]: In the examples we can use tailwind css.

File: src/App.tsx (hunk 1 above)
Comment (ankit-adapt) [reply src/App.tsx]: Fixed

**Pull request closed without merging**